sdk/src/main/scala/com/horizen/account/receipt/EthereumReceipt.scala

```diff
@@ -128,10 +144,10 @@ object EthereumReceiptSerializer extends SparkzSerializer[EthereumReceipt]{
     val gasUsedLength: Int = reader.getInt
     val gasUsed: BigInteger = new BigInteger(reader.getBytes(gasUsedLength))
 
-    //optional field
+    // optional field
```

Comment:
here we should refactor and serialize/deserialize  `contractAddress` as Option.
So in case the receipt is not for a smart contract declaration -> `None`.

Otherwise, what is the sense of using Option type?

Reply:
@GitMark0 This is still missing. Read the length, if zero contractAddress should be `None`.

Reply:
fxied